synapse/rest/client/shared_rooms.py

```diff
@@ -48,13 +48,6 @@ async def on_GET(
         self, request: SynapseRequest, user_id: str
     ) -> Tuple[int, JsonDict]:
 
-        if not self.user_directory_active:
```

Comment:
This check probably wants to be replaced with `self.hs.config.userdirectory.user_directory_search_enabled` as we do for `/user_directory/search`:

https://github.com/matrix-org/synapse/blob/9e83521af860cb33a7459dbe74188ce5ef39f446/synapse/rest/client/user_directory.py#L61-L62

Reply:
No, please review this comment; https://github.com/matrix-org/synapse/issues/10339#issuecomment-1039175679

TLDR; This PR explicitly deals with the removal of this line, the only thing this bit of code needs is for the user directory updater to exist *somewhere* in the cluster, and by default it'll exist on the main process.

The pros and cons of this have already been agreed upon by anoa and rich in that issue.

Reply:
> TLDR; This PR explicitly deals with the removal of this line, the only thing this bit of code needs is for the user directory updater to exist _somewhere_ in the cluster, and by default it'll exist on the main process.

if `self.hs.config.userdirectory.user_directory_search_enabled` is `False` then the user directory updater doesn't exist anywhere, as its completely disabled. The issue seems to be only discussing the worker case, rather than the user directory being disabled entirely? I don't really understand why we would want to process the request if the directory is disabled, we should either `400` or early return with an empty response like we do for the search endpoint.

Reply:
I'm happy with 400ing and changing if the MSC decides something different :+1: